clients/drcachesim/tests/burst_flush_aarch64.cpp

```diff
@@ -37,39 +38,73 @@
 #include <assert.h>
 #include <iostream>
 #include <signal.h>
+#include <setjmp.h>
+
+static sigjmp_buf mark;
+static int handled_sigill_count = 0;
 
 bool
 my_setenv(const char *var, const char *value)
 {
-#ifdef UNIX
     return setenv(var, value, 1 /*override*/) == 0;
-#else
-    return SetEnvironmentVariable(var, value) == TRUE;
-#endif
 }
 
 void
 sigill_handler(int signal)
 {
-    dr_app_stop_and_cleanup();
-    std::cerr << "all done\n";
-    _Exit(0);
+    handled_sigill_count++;
+    siglongjmp(mark, handled_sigill_count);
+    exit(-1);
 }
 
 static void
-do_some_work()
+dc_ivac()
 {
     int d = 0;
-    // Expected to send SIGILL.
+    // Expected to raise SIGILL.
     // Control will be transferred to sigill_handler
-    __asm__ __volatile__("dc ivac, %0" : : "r"(d));
+    __asm__ __volatile__("dc ivac, %0" : : "r"(&d));
+}
+
+static void
+dc_unprivileged_flush()
+{
+    int d = 0;
+    __asm__ __volatile__("dc civac, %0" : : "r"(&d));
+    __asm__ __volatile__("dc cvac , %0" : : "r"(&d));
+    __asm__ __volatile__("dc cvau , %0" : : "r"(&d));
+}
+
+static void
+ic_unprivileged_flush()
+{
+    __asm__ __volatile__("ic ivau , %0" : : "r"(ic_unprivileged_flush));
+}
+
+static void
+do_some_work()
+{
+    dc_unprivileged_flush();
+    ic_unprivileged_flush();
+
+    // Testing privileged instructions requires handling SIGILL.
+    // We use sigsetjmp/siglongjmp to resume execution after handling the
+    // signal.
+    handled_sigill_count = 0;
+    int i = sigsetjmp(mark, 1);
+    switch (i) {
```

Comment:
Either have a TODO comment about adding more cases, or this should just be an `if` for just one case.

Reply:
Added a TODO referencing #4406.